presto-memory/src/test/java/io/prestosql/plugin/memory/TestMemorySmoke.java

```diff
@@ -263,6 +264,61 @@ public void testSemiJoinDynamicFilteringBlockProbeSide()
+        assertUpdate("INSERT INTO build VALUES (1, 2)", 1);
+
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v <= vmax", withBroadcastJoin(), 3, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v < vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v < vmax", withBroadcastJoin(), 1, ImmutableSet.of(1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND v <= vmax", withBroadcastJoin(), 1, ImmutableSet.of(1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND v < vmax", withBroadcastJoin(), 0, ImmutableSet.of(0, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v > vmin AND vmax < 0", withBroadcastJoin(), 0, ImmutableSet.of(0, 1));
+
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin AND vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v <= vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe, build WHERE v BETWEEN vmin AND vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+        assertDynamicFiltering("SELECT * FROM probe, build WHERE v >= vmin AND v <= vmax", withBroadcastJoin(), 2, ImmutableSet.of(2, 1));
+
+        // TODO: support complex inequality join clauses: https://github.com/prestosql/presto/issues/5755
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin AND vmax - 1", withBroadcastJoin(), 1, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin + 1 AND vmax", withBroadcastJoin(), 1, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v BETWEEN vmin + 1 AND vmax - 1", withBroadcastJoin(), 0, ImmutableSet.of(4, 1));
+        assertDynamicFiltering("SELECT * FROM probe, build WHERE v BETWEEN vmin AND vmax - 1", withBroadcastJoin(), 1, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe, build WHERE v BETWEEN vmin + 1 AND vmax", withBroadcastJoin(), 1, ImmutableSet.of(3, 1));
+        assertDynamicFiltering("SELECT * FROM probe, build WHERE v BETWEEN vmin + 1 AND vmax - 1", withBroadcastJoin(), 0, ImmutableSet.of(4, 1));
+
+        // TODO: make sure it works after https://github.com/prestosql/presto/issues/5777 is fixed
+        assertDynamicFiltering("SELECT * FROM probe JOIN build ON v >= vmin AND v <= vmax - 1", withBroadcastJoin(), 1, ImmutableSet.of(1));
```

Comment:
why set has one element?

Reply:
You're right, it's better to use `ImmutableList.of(1, 1)` here.